torch/_dynamo/eval_frame.py

```diff
@@ -670,6 +672,11 @@ def guard_export_print(guards: Set[_guards.Guard]):
         assert out_guards is None, "whole graph export entails exactly one guard export"
         out_guards = guards
 
+    def expose_shape_env_fn(shape_env: ShapeEnv):
+        nonlocal out_shape_env
+        assert out_shape_env is None, "whole graph should only have one shape env"
+        out_shape_env = shape_env
```

Comment:
Instead of this, when you are called in dynamo_normalization_capturing_compiler your example inputs will be fake tensors, whose fake tensor mode will have shape env on it. Extract the shape env that way. This is how AOTAutograd does it, by the way.

Reply:
I see. that sounds easier